crates/autopilot/src/database/order_events.rs

```diff
@@ -18,21 +19,29 @@ impl super::Postgres {
 }
 
 pub async fn store_order_events(
-    db: &super::Postgres,
+    ex: &mut PgConnection,
     order_uids: Vec<domain::OrderUid>,
     label: OrderEventLabel,
     timestamp: DateTime<Utc>,
 ) -> Result<()> {
-    let mut ex = db.pool.begin().await.context("begin transaction")?;
+    let start = Instant::now();
+    let events_count = order_uids.len();
     for uid in order_uids {
         let event = OrderEvent {
             order_uid: ByteArray(uid.0),
             timestamp,
             label,
         };
 
-        order_events::insert_order_event(&mut ex, &event).await?
+        match order_events::insert_order_event(ex, &event).await {
+            Ok(_) => {
+                tracing::debug!(elapsed=?start.elapsed(), ?events_count, "stored order events");
```

Comment:
You are printing the log for each order in the loop (claiming that it stored `events_count` things).
Over all I would say `store_order_events` should:
* take a `&mut PgConnection`
* start a fresh transaction
* handle errors internally by logging
* return nothing (an error in this logic is not critical and already got logged internally)

Then the caller does not have to care about transactions or the function throwing errors at all making all call sites nicer to read without compromising anything (since we didn't handle any errors before anyway).

Reply:
There is still one log like `events_count=12 stored order events` for every event that gets inserted.
What makes more sense would be something like `order_count=10 stored order events` and `order_count=3 failed to store order events` (once instead of for every single insert).